lib/cext/preprocess.rb

```diff
@@ -783,6 +783,103 @@ def tuple_new_patch(ctx, slf)
+      },
+      { # trace
+        match: 'sqlite3_trace(ctx->db, NIL_P(block) ? NULL : tracefunc, (void *)self);',
+        replacement: 'sqlite3_trace(ctx->db, NIL_P(block) ? NULL : tracefunc, rb_tr_handle_for_managed_leaking(self));'
+      },
+      { # rb_sqlite3_busy_handler
+        match: 'VALUE self = (VALUE)(ctx);',
+        replacement: 'VALUE self = rb_tr_managed_from_handle(ctx);'
+      },
+      { # busy_handler
+        match: 'ctx->db, NIL_P(block) ? NULL : rb_sqlite3_busy_handler, (void *)self);',
+        replacement: 'ctx->db, NIL_P(block) ? NULL : rb_sqlite3_busy_handler, rb_tr_handle_for_managed_leaking(self));'
+      },
+      { # rb_sqlite3_func
+        match: 'VALUE callable = (VALUE)sqlite3_user_data(ctx);',
+        replacement: 'VALUE callable = rb_tr_managed_from_handle(sqlite3_user_data(ctx));'
+      },
+      { # define_function_with_flags
+        match: 'NUM2INT(flags),',
+        replacement: 'flags,'
+      },
+      { # define_function_with_flags
+        match: '(void *)block,',
+        replacement: 'rb_tr_handle_for_managed_leaking(block),'
+      },
+      { # rb_sqlite3_step
+        match: 'VALUE callable = (VALUE)sqlite3_user_data(ctx);',
+        replacement: 'VALUE callable = rb_tr_managed_from_handle(sqlite3_user_data(ctx));'
+      },
+      { # rb_sqlite3_step
+        match: 'params = xcalloc((size_t)argc, sizeof(VALUE *));',
+        replacement: 'params = truffle_managed_malloc((size_t)argc * sizeof(VALUE *));'
+      },
+      { # rb_sqlite3_step
+        match: 'xfree(params);',
+        replacement: ''
+      },
+      { # rb_sqlite3_final
+        match: 'VALUE callable = (VALUE)sqlite3_user_data(ctx);',
+        replacement: 'VALUE callable = rb_tr_managed_from_handle(sqlite3_user_data(ctx));'
+      },
+      { # define_aggregator
+        match: '(void *)aggregator,',
+        replacement: 'rb_tr_handle_for_managed_leaking(aggregator),'
+      },
+      { # rb_sqlite3_auth
+        match: 'VALUE self   = (VALUE)ctx;',
+        replacement: 'VALUE self   = rb_tr_managed_from_handle(ctx);'
+      },
+      { # rb_sqlite3_auth
+        match: 'if(Qfalse == result) return SQLITE_DENY;',
+        replacement: 'if(rb_tr_obj_equal(Qfalse, result)) return SQLITE_DENY;'
+      },
+      { # set_authorizer
+        match: 'ctx->db, NIL_P(authorizer) ? NULL : rb_sqlite3_auth, (void *)self',
+        replacement: 'ctx->db, NIL_P(authorizer) ? NULL : rb_sqlite3_auth, rb_tr_handle_for_managed_leaking(self)'
+      },
+      { # rb_comparator_func
+        match: 'comparator = (VALUE)ctx;',
+        replacement: 'comparator = rb_tr_managed_from_handle(ctx);'
+      },
+      { # collation
+        match: '(void *)comparator,',
+        replacement: 'rb_tr_handle_for_managed_leaking(comparator),'
+      },
+      { # enc_cb
+        match: 'rb_iv_set(self, "@encoding", rb_enc_from_encoding(e));',
+        replacement: 'rb_iv_set(rb_tr_managed_from_handle(_self), "@encoding", rb_enc_from_encoding(e));'
```

Comment:
I there supposed to be underscore here? `_self`

Reply:
Yes. That's the parameter name to the `enc_cb` function. I could move the managed handle creation elsewhere though and make the name `self`

Reply:
I've updated the patch to move the call location and better match the other patches here.